### PR TITLE
Read & Upload media files  https://github.com/asmecher/texture/issues/2

### PR DESCRIPTION
**Pull request by withanage**
@asmecher 

This pull request  addresses the file read and upload issue for both master and ojs-stable-3_1_1

https://github.com/asmecher/texture/issues/2

* Added separate request handling for PUT, GET 
* for dependent files  I added some two hard-corded variables.  I was  not sure, if there is a better way for that. 
```
$genreId = 10; /
$fileStage = 17;
```
https://github.com/withanage/texture/commit/b991239f860dbc5dd92eef26e2b49164a4c79ca6#diff-ef32d535c698ff4d800e1bbc57f3c937R150

### CODE REVIEW

**Comment by asmecher** [comment]
(See my notes on the hard-coded constants over here: https://github.com/asmecher/texture/pull/14#issuecomment-447932482)

**Comment by withanage** [comment]
@asmecher 

Delete file functionality for master branch

https://github.com/asmecher/texture/pull/14#issuecomment-449129937

File: TextureHandler.inc.php [hunk 1]
```diff
@@ -15,14 +15,16 @@
...
+class TextureHandler extends Handler
+{
 	/** @var MarkupPlugin The Texture plugin */
 	protected $_plugin;
 
 	/**
 	 * Constructor
 	 */
-	function __construct() {
+	function __construct()
```
Comment (asmecher) [inline TextureHandler.inc.php]: For consistency with the rest of the codebase, stick with [K&R style brace indentation](https://en.wikipedia.org/wiki/Indentation_style#Brace_placement_in_compound_statements).

File: TextureHandler.inc.php [hunk 2]
```diff
@@ -85,143 +90,142 @@ public function editor($args, $request) {
...
+					'size' => filesize($document->path),
+					'createdAt' => 0,
+					'updatedAt' => 0,
 				),
 			);
-			$data = array(
-				'version'       => 'AE2F112D',
-				'resources'     => array_merge($resources, $mediaInfos)
+			$mediaBlob = array(
+				'version' => 'AE2F112D',
```
Comment (asmecher) [inline TextureHandler.inc.php]: Just curious -- what's this hash?

**Comment by asmecher** [comment]
Looks good, @withanage -- please go ahead and merge, and I'll be on hand for some testing & release work!

File: TextureHandler.inc.php (hunk 1 above)
Comment (withanage) [reply TextureHandler.inc.php]: sorry, @asmecher  for that. 
I have reset the functional braces and class braces in the file and formatted accordingly.

File: TextureHandler.inc.php (hunk 2 above)
Comment (withanage) [reply TextureHandler.inc.php]: The version hash was already there, I think it was just added for compatibility, so I let it.
Cause texture 1.0 client does not use it,  now I replaced it wit the current XML submission file version ,then I can use  this version number to detect changes in the OJS Backend version of the XML file.

**Comment by withanage** [comment]
thanks @asmecher 

Now all the basic functionality is ready for going ahead wit some internal user tests. 

Just one thing  which is missing : 
If users redo and undo the images using texture redo/undo, texture does not send the new images in the manifest. Therefore the image does not go to the server. 
I think it is only a not-yet implemented feature in texture and they will build  it in a later release.  
I can  dig a  into texture  javascript actions and add a  hook kind of functionality to  get those changes too, but my worry is , if they implement is later, then the work is done twice.

What do yo think best ?

**Comment by asmecher** [comment]
@withanage, I'd suggest filing an issue on the Texture repo; that seems like something that needs to be solved on their side. Meanwhile, please go ahead and merge!

**Pull request merged**